solution03/src/clients/createBudget.js

```diff
@@ -0,0 +1,25 @@
+import Budget from '../models/budget.js'
+
+export default async (budgetParams) => {
+
+    const checkIfBudget = await Budget.find()
```

Comment:
To implement locks you need first to use a library like locks.
```js
import locks from "locks";

```

Then lock the transactions:
`const mutex = locks.createMutex();`

And use a mutex function like:
```js
mutex.lock(async () => { 
... code 
mutex.unlock(); // don't forget to unlock the file
})
```